src/DynamoCore/Core/Threading/AsyncTaskExtensions.cs

```diff
@@ -7,24 +7,101 @@
 
 namespace Dynamo.Core.Threading
 {
+    /// <summary>
+    ///     Tools for working productively with AsyncTask's
+    /// </summary>
     internal static class AsyncTaskExtensions
     {
+        /// <summary>
+        ///     Upon completion of the task, invoke the specified action
+        /// </summary>
+        /// <returns>An IDisposable representing the event subscription</returns>
         internal static IDisposable Then(this AsyncTask task, AsyncTaskCompletedHandler action)
         {
             task.Completed += action;
             return Disposable.Create(() => task.Completed -= action);
         }
 
+        /// <summary>
+        ///     Upon completion of the task, invoke the specified asynchronously in the specified
+        ///     SynchronizationContext
+        /// </summary>
+        /// <returns>An IDisposable representing the event subscription</returns>
         internal static IDisposable ThenPost(this AsyncTask task, AsyncTaskCompletedHandler action, SynchronizationContext context = null)
```

Comment:
I don't think the name of this function is helpful - could we consider renaming. This became more obvious with your comment.